addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2287,6 +2287,22 @@ const SnippetOptionWidget = Widget.extend({
         const value = this._selectAttributeHelper(widgetValue, params);
         this.$target[0].setAttribute(params.attributeName, value);
     },
+    /**
+     * Default option method which allows to select a value and set it on the
+     * associated snippet as a property. The name of the property is
+     * given by the propertyName parameter.
+     *
+     * @param {boolean} previewMode - @see this.selectClass
+     * @param {string} widgetValue
+     * @param {Object} params
+     */
+    selectProperty: function (previewMode, widgetValue, params) {
+        if (!params.propertyName) {
+            throw new Error('Property name missing');
+        }
+        const value = this._selectValueHelper(widgetValue, params);
+        this.$target[0][params.propertyName] = value;
+    },
```

Comment:
@qsm-odoo do we need this when we already have a `selectAttribute` method? In which cases does the behaviour differ enough to justify a separate method?

Reply:
nevermind. `input.value`, which will change the default value when using `setAttribute` but will change the current value when using direct property access.